docs/root/intro/arch_overview/upstream/connection_pooling.rst

```diff
@@ -32,17 +32,57 @@ pool will drain the affected connection. Once a connection reaches its :ref:`max
 stream limit <envoy_v3_api_field_config.core.v3.Http2ProtocolOptions.max_concurrent_streams>`, it
 will be marked as busy until a stream is available. New connections are established anytime there is
 a pending request without a connection that can be dispatched to (up to circuit breaker limits for
-connections). HTTP/2 is the preferred communication protocol, as connections rarely, if ever, get
-severed.
+connections). HTTP/2 is the preferred communication protocol when Envoy is operating as a reverse proxy,
+as connections rarely, if ever, get severed.
+
+HTTP/3
+------
+
+The HTTP/3 connection pool multiplexes multiple requests over a single connection, up to the limits
+imposed by :ref:`max concurrent streams
+<envoy_v3_api_field_config.core.v3.QuicProtocolOptions.max_concurrent_streams>` and :ref:`max
+requests per connection <envoy_v3_api_field_config.cluster.v3.Cluster.max_requests_per_connection>`.
+The HTTP/3 connection pool establishes as many connections as are needed to serve requests. With no
+limits, this will be only a single connection. If a GOAWAY frame is received or if the connection
+reaches the :ref:`maximum requests per connection
+<envoy_v3_api_field_config.cluster.v3.Cluster.max_requests_per_connection>` limit, the connection
+pool will drain the affected connection. Once a connection reaches its :ref:`maximum concurrent
+stream limit <envoy_v3_api_field_config.core.v3.QuicProtocolOptions.max_concurrent_streams>`, it
+will be marked as busy until a stream is available. New connections are established anytime there is
+a pending request without a connection that can be dispatched to (up to circuit breaker limits for
+connections). HTTP/3 upstream support is currently only usable in situations where HTTP/3 is guaranteed
+to work, but automatic failover to TCP is coming soon!.
+
+Automatic protocol selection
+----------------------------
+
+For Envoy acting as a forward proxy, the preferred configuration is the
+`AutoHttpConfig <envoy_v3_api_msg_extensions.upstreams.http.v3.HttpProtocolOptions.AutoHttpConfig>`
+, configued via
+`http_protocol_options <envoy_v3_api_msg_extensions.upstreams.http.v3.HttpProtocolOptions>`.
+By default it will use TCP and ALPN to select the best available protocol of HTTP/2 and HTTP/1.1.
+
+.. _arch_overview_http3_upstream:
+
+If HTTP/3 is configured in the automatic pool, it will currently attempt an QUIC connection first,
+then 300ms later, attempt a TCP connection. Whichever handshake succeeds will be used for the initial
```

Comment:
Maybe this doesn't need to be mentioned, but TCP is only attempted after the timeout if the QUIC connection has not already succeeded, right?